src/tests/architect_tests/test_integration.py

```diff
@@ -356,9 +361,11 @@ def basic_integration_test(
             assert len(matrices) == num_split_matrices * expected_matrix_multiplier
             assert len(metadatas) == num_split_matrices * expected_matrix_multiplier
 
+            return matrices_records
```

Comment:
I think if you add an `assert len(matrices) == len(matrices_records)` here you can avoid doing length checks in the individual tests. In fact, as a sanity check you could compare many other properties of the matrices on disk (say from the metadata) to what ended up in the table.